eng/Build.proj

```diff
@@ -1,37 +1,55 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information. -->
-<Project DefaultTargets="ClearNuGetCache;PrepareMachine;CoreBuild">
```

Comment:
ClearNuGetCache was deliberate to avoid taking a dependency on packages that had been deleted, or when packages sources changed.

Reply:
That target only ran if you provided the `/clearnugetcache` as a parameter to *build.cmd* or `/p:ClearNuGetCache=false` to *Build.proj*. It wasn't being used in CI and there wasn't a *clear* need for it as part of our build process. As I mentioned [here](https://github.com/dotnet/project-system/pull/8353/files#r931696942), it was simply deleting all the packages our of your Windows user cache. If this repo had a local cache, such as a `packages` folder that is created at the repo root on restore, I could see it being nice to have a flag to clear that and maybe even tying that into Rebuild.

As mentioned, if there is a scenario where it is necessary, I haven't ran into it yet, but I can add back in an equivalent. Generally, clearing out your *entire* cache shouldn't be necessary; it is more in exceptional situations for resolving specific issues. For the scenario you mentioned, we don't take on many new dependencies at this point, and I did a full sweep of package source feeds recently, so I don't feel a strong need for this flag/target.

Reply:
Here's a good example of a current PR where, maybe your cache on your machine had the right packages. However, CI run will always fail if the package dependencies are not defined properly. Drew can test it locally by clearing his cache, but this should not be seen as a normal scenario. https://github.com/dotnet/project-system/pull/8346